commands/utils/packagehandlers/mavenpackagehandler.go

```diff
@@ -1,42 +1,324 @@
+	mavenDepToPropertyMap map[string]pomDependencyDetails
+	// pomPaths holds the paths to all the pom.xml files that are related to the current project.
+	pomPaths []pomPath
+	// A flag that determines whether the maven-gav-reader.jar plugin should be installed or if it is already installed.
+	isMavenGavReaderInstalled bool
+	// The server details for Artifactory in case of an air-gapped environment.
+	*config.ServerDetails
+	// The remote repository in Artifactory to resolve dependencies from.
+	depsRepo string
 }
 
-func (mvn *MavenPackageHandler) UpdateImpactedPackage(impactedPackage string, fixVersionInfo *utils.FixVersionInfo, extraArgs ...string) error {
-	// In Maven, fix only direct dependencies
-	if !fixVersionInfo.DirectDependency {
+func (mph *MavenPackageHandler) UpdateImpactedPackage(impactedPackage string, fixVersionInfo *utils.FixVersionInfo, _ ...string) error {
+	if err := mph.installMavenGavReader(); err != nil {
+		return err
+	}
+	if err := mph.getProjectPoms(); err != nil {
+		return err
+	}
+	// Get direct dependencies for each pom.xml file
+	if mph.mavenDepToPropertyMap == nil {
+		mph.mavenDepToPropertyMap = make(map[string]pomDependencyDetails)
+	}
+	for _, pp := range mph.pomPaths {
+		if err := mph.fillDependenciesMap(pp.PomPath); err != nil {
+			return err
+		}
+	}
+
+	var depDetails pomDependencyDetails
+	var exists bool
+	// Check if the impacted package is a direct dependency
+	if depDetails, exists = mph.mavenDepToPropertyMap[impactedPackage]; !exists {
+		return fmt.Errorf(utils.SkipIndirectVulnerabilitiesMsg, impactedPackage, fixVersionInfo.FixVersion)
+	}
+
+	if len(depDetails.properties) > 0 {
+		return mph.updateProperties(&depDetails, fixVersionInfo.FixVersion)
+	}
+
+	return mph.updatePackageVersion(impactedPackage, fixVersionInfo.FixVersion, depDetails.foundInDependencyManagement)
+}
+
+func (mph *MavenPackageHandler) installMavenGavReader() (err error) {
+	if mph.isMavenGavReaderInstalled {
 		return nil
 	}
-	properties := mvn.mavenDepToPropertyMap[impactedPackage]
-	// Update the package version. This command updates it only if the version is not a reference to a property.
-	updateVersionArgs := []string{"-B", "versions:use-dep-version", "-Dincludes=" + impactedPackage, "-DdepVersion=" + fixVersionInfo.FixVersion, "-DgenerateBackupPoms=false"}
-	updateVersionCmd := fmt.Sprintf("mvn %s", strings.Join(updateVersionArgs, " "))
-	log.Debug(fmt.Sprintf("Running '%s'", updateVersionCmd))
-	updateVersionOutput, err := exec.Command("mvn", updateVersionArgs...).CombinedOutput() // #nosec G204
+	// Create maven-gav-reader plugin file and write the maven-gav-reader.jar content to it
+	mavenGavReaderFile, err := os.CreateTemp("", fmt.Sprintf("*-%s", mavenGavReader))
+	if err != nil {
+		return fmt.Errorf("failed to create a temp %s file: \n%s", mavenGavReader, err.Error())
+	}
+	defer func() {
+		e1 := mavenGavReaderFile.Close()
+		e2 := os.Remove(mavenGavReaderFile.Name())
```

Comment:
Prefer informative variable names
```suggestion
		closeError := mavenGavReaderFile.Close()
		deleteError := os.Remove(mavenGavReaderFile.Name())
```